src/sagemaker_sklearn_extension/feature_extraction/sequences.py

```diff
@@ -0,0 +1,414 @@
+    max_allowed_length : int (default = 10000)
+        Maximum allowed length of an input sequence. If the length of a sequence is greater than ``max_allowed_length``,
+        the transformer will truncate its beginning or end as indicated by ``trim_beginning``.
+
+    trim_beginning : bool (default = True)
+        If a sequence length exceeds ``max_allowed_length``, trim its start and only keep the last `max_allowed_length``
+        values if ``trim_beginning`` = True, otherwise trim the end and only keep the first max_allowed_length values.
+
+
+    Examples
+    --------
+    >>> from sagemaker_sklearn_extension.feature_extraction.sequences import TSFlattener
+    >>> import numpy as np
+    >>> data = [["1,, 3, 44"], ["11, 111"], ["NaN, , 1, NaN"]]
+    >>> ts_flattener = TSFlattener()
+    >>> X = ts_flattener.transform(data)
+    >>> print(len(X))
+    3
+    >>> print(X)
+    [array([ 1., nan,  3., 44.]), array([ 11., 111.]), array([nan, nan,  1., nan])]
+    >>> ts_flattener = TSFlattener(max_allowed_length=2, trim_beginning=True)
+    >>> X = ts_flattener.transform(data)
+    >>> print(X)
+    [array([ 3., 44.]), array([ 11., 111.]), array([ 1., nan])]
+    """
+
+    def __init__(self, max_allowed_length=10000, trim_beginning=True):
+        super().__init__()
+        assert max_allowed_length > 0, f"{max_allowed_length} must be positive."
+        self.max_allowed_length = max_allowed_length
+        self.trim_beginning = trim_beginning
+
+    def fit(self, X, y=None):
+        check_array(X, dtype=None, force_all_finite="allow-nan")
+        return self
+
+    def transform(self, X, y=None):
+        """Extract numerical values from strings of comma-separated numbers and returns an np.array.
+
+        Anything that can't be turned into a finite float is converted to a np.nan.
+
+        Parameters
+        ----------
+        X : lists of strings
+
+        Returns
+        -------
+        X : List of np.arrays
+
+        """
+        X = check_array(X, dtype=None, force_all_finite="allow-nan")
+        # Parse the input strings
+        numeric_sequences = self._convert_to_numeric(X)
+        return numeric_sequences
+
+    def _convert_to_numeric(self, X):
+        numeric_sequences = []
+        for string_sequence in X:
+            assert len(string_sequence) == 1, (
+                f"TSFlattener can process a single sequence column at a time, "
+                f"but it was given {len(string_sequence)} sequence columns."
+            )
+            numeric_sequence = []
+            if string_sequence[0] is not None:
+                for s in string_sequence[0].split(","):
+                    # Turn anything that can't be converted to a finite float to np.nan
+                    try:
+                        s = float(s)
+                    except ValueError:
+                        s = np.nan
+                    if np.isinf(s):
+                        s = np.nan
+                    numeric_sequence.append(s)
```

Comment:
This could be very computationally expensive. Do we have benchmarks on larger datasets?

Reply:
We benchmarked on the popular set of UCR datasets for time series classification (https://www.cs.ucr.edu/~eamonn/time_series_data_2018/) and didn't notice runtime issues there. I am not sure how this parsing should be changed to be more efficient.